js/components/graph/GraphDropdown.js

```diff
@@ -0,0 +1,104 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+export default class GraphDropdown extends React.Component{
+
+	generateDropdownItems(column){
+		return column.map((graph, i) => {
+			let titleStart = graph.title.lastIndexOf(")");
+			let graphDisplayTitle = (titleStart === -1) ? graph.title : graph.title.slice(titleStart + 2);
+			return <li
+							key={i}
+							className="graph-dropdown-item"
+							onClick={() => this.props.updateGraph(graph.title)}
+							data-testid={"test-graph-" + i}
+							>
+								{graphDisplayTitle}
+							</li>
+		})
+	}
+
+	createDropdown(dropdownClassName, leftOffset){
+		let officialColumn = [];
+		let businessColumn = [];
+		let sciencesColumn = [];
+		let socialStudiesColumn = [];
+		let languagesColumn = [];
+		let graphToColumnMapping = {
+			"Computer Science": officialColumn,
+			"(unofficial) Rotman": businessColumn,
+			"(unofficial) Statistics": sciencesColumn,
+			"(unofficial) Biochemistry": sciencesColumn,
+			"(unofficial) Cell & Systems Biology": sciencesColumn,
+			"(unofficial) Linguistics": sciencesColumn,
+			"(unofficial) History and Philosophy of Science": socialStudiesColumn,
+			"(unofficial) History": socialStudiesColumn,
+			"(unofficial) Geography": socialStudiesColumn,
+			"(unofficial) Aboriginal": socialStudiesColumn,
+			"(unofficial) East Asian Studies": socialStudiesColumn,
+			"(unofficial) Economics": socialStudiesColumn,
+			"(unofficial) Finnish": languagesColumn,
+			"(unofficial) Italian": languagesColumn,
+			"(unofficial) Spanish": languagesColumn,
+			"(unofficial) Portuguese": languagesColumn,
+			"(unofficial) German": languagesColumn,
+			"(unofficial) Estonian": languagesColumn,
+			"(unofficial) Slavic": languagesColumn,
+			"(unofficial) English": languagesColumn
+		}
+
+		this.props.graphs.forEach((graph) => {
+			graphToColumnMapping[graph.title].push(graph)
+		});
+
+		return (
+		<ul
+			className={dropdownClassName}
+			style={{left:leftOffset}}
+			onMouseMove={this.props.onMouseMove}
+			onMouseLeave={this.props.onMouseLeave}
+			data-testid={"test-graph-dropdown"}
+		>
+			<div>
+				<h4>Official Graphs</h4>
+				{this.generateDropdownItems(officialColumn)}
+			</div>
+			<div>
+				<h4>(Unofficial) Sciences</h4>
+				{this.generateDropdownItems(sciencesColumn)}
+			</div>
+			<div>
+				<h4>(Unofficial) Business</h4>
+				{this.generateDropdownItems(businessColumn)}
+			</div>
+			<div>
+				<h4>(Unofficial) Social Studies</h4>
+				{this.generateDropdownItems(socialStudiesColumn)}
+			</div>
+			<div>
+				<h4>(Unofficial) Languages</h4>
+				{this.generateDropdownItems(languagesColumn)}
+			</div>
+		</ul>
+
+		);
+	}
+
+	render() {
+		let halfComponentWidth = 412;
```

Comment:
These numbers shouldn't be hard-coded. The `Graph` component computes its width, and you can pass that information as a prop to this component.

For `graphTabLeft`, there again should not be a default number. If the nav bar item can't be found, it's better not to display anything rather than display a component that doesn't make sense.

Reply:
The halfComponentWidth is based off half the width of the GraphDropdown component. I couldn't figure out how to get GraphDropdown centered on the graph tab in css. 

I think with the text aligned to the left it looks good without trying to center GraphDropdown on the graph tab (screenshot below). Are you ok with that? If not, I think there's the translation property in CSS which I could use to center the graph dropdown, so I can just do that.

![image](https://user-images.githubusercontent.com/22108651/126048003-3b14c557-f2e4-4ee6-96bd-d846858e907b.png)

Reply:
@kenneth-miura oh I see, yes I think aligning it to the left looks good, you can stick with that!

(BTW, make sure to add some padding to the dropdown container, so that the text isn't right against the borders).